File: doc-examples/jdbc-example-java/src/main/java/example/StudentScheduleSubView.java
package example;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.JsonSubView;
import io.micronaut.data.annotation.JsonView;

// tag::class-example[]
@JsonSubView(entity = StudentClass.class, operations = { JsonView.Operation.UPDATE, JsonView.Operation.INSERT })
public class StudentScheduleSubView {
// end::class-example[]
    @Id
    private Long id;
}

File: doc-examples/jdbc-example-python/src/main/python/example/StudentScheduleSubView.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import Id, JsonSubView

from example.StudentClass import StudentClass


# tag::class-example[]
@JsonSubView(entity=StudentClass, operations=["UPDATE", "INSERT"])
@dataclass
class StudentScheduleSubView:
    # end::class-example[]
    id: Annotated[int | None, Id] = None
